src/actions/filterActions.js

```diff
@@ -1,88 +1,102 @@
+
+export const applyEventTypeAddition = (history, oldFilter, signalRFilterType, eventType) => (dispatch) => {
+  if (nothingToAdd(eventType) || typeIsAlreadyFiltered(oldFilter, eventType)) {
     return
   }
-  if (oldFilter && oldFilter.eventTypes && oldFilter.eventTypes.includes(eventType.id)) {
-    newFilter = { ...oldFilter }
-  } else {
-    newFilter = {
-      ...oldFilter,
-      eventTypes: oldFilter.eventTypes
-        ? oldFilter.eventTypes.concat(eventType.id)
-        : [eventType.id]
-    }
-  }
-  return applyFilter(history)(oldFilter, newFilter, signalRFilterType)
+  dispatch(addEventTypeToFilter(eventType.id))
+
+  const newFilter = addTypeToOldFilter(oldFilter, eventType)
+
+  dispatch(applyFilter(history, newFilter, signalRFilterType))
 }
 
-export const removeFilter = (history, relativeFilterPath) => (oldFilter, filterToDelete) => {
-  if (!ByPath.get(oldFilter, relativeFilterPath).includes(filterToDelete)) {
+const nothingToRemove = (oldFilter, eventTypeId) => !oldFilter.eventTypes || !oldFilter.eventTypes.includes(eventTypeId)
+
+export const applyEventTypeRemoval = (history, signalRFilterType, oldFilter, eventTypeId) => (dispatch) => {
+  if (nothingToRemove(oldFilter, eventTypeId)) {
     return
   }
-  let newFilter = { ...oldFilter }
-  const isFilterToKeep = existingFilter => filterToDelete !== existingFilter
-  ByPath.set(newFilter, relativeFilterPath, ByPath.get(oldFilter, relativeFilterPath).filter(isFilterToKeep))
-  return applyFilter(history)(oldFilter, newFilter)
-}
+  dispatch(removeEventTypeFromFilter(eventTypeId))
 
-const applyFilter = (history) => (oldFilter, newFilter, signalRFilterType) => (dispatch) => {
-  if (newFilter.incidentId) {
-    if (!deepEquals(oldFilter, newFilter)) {
-      dispatch(signalRActions.updateEventFilterPreference(signalRFilterType, newFilter))
-      dispatch(changeEventFilter(history, filterService.getUrlFromFilter)(newFilter))
-      dispatch(eventActions.fetchEvents(newFilter))
-    }
-  } else {
-    dispatch(signalRActions.updateEventFilterPreference(signalRFilterType, newFilter))
-    dispatch(changeEventFilter(history, filterService.getUrlFromUncorrelatedFilter)(newFilter))
-    dispatch(eventActions.fetchUncorrelatedEvents(newFilter))
+  const isFilterToKeep = existingId => eventTypeId !== existingId
+  const newFilter = {
+    ...oldFilter,
+    eventTypes: oldFilter.eventTypes.filter(isFilterToKeep)
   }
+
+  dispatch(applyFilter(history, newFilter, signalRFilterType))
 }
 
-export const synchronizeFilters = (filter, incidentId, ticketId, history) => {
-  const newFilter = Object.assign({}, filter, { incidentId: incidentId, ticketId: ticketId })
-  return applyFilter(history)(filter, newFilter)
+export const applyFilter = (history, filter, signalRFilterType) => (dispatch) => {
+  filterService.updateUrlToMatchFilter(history, filter)
+  dispatch(signalRActions.updateEventFilterPreference(signalRFilterType, filter))
+  dispatch(eventActions.clearEvents())
```

Comment:
Why do you clear the current events before fetching new events?

Reply:
Because that clears out events that don't match the newly applied filter.

Reply:
We were not clearing out old events previously

Reply:
Previously, CHANGE_EVENT_FILTER cleared out events. Check the EventReducer. The effect of that was that whenever the filter changed, the event collection was dropped.